rosetta-source/src/main/rosetta/product-collateral-type.rosetta

```diff
@@ -157,44 +155,80 @@ type ConcentrationLimitCriteria extends CollateralCriteriaBase: <"Respresents a
+type AssetCountryOfOrigin:
+    assetCountryOfOrigin ISOCountryCodeEnum (1..1) <"Represents a filter on the asset country of origin based on the ISO Standard 3166.">
 
-type AssetCriteria: <"Represents a set of criteria used to specify eligible collateral assets.">
+type IssuerName:
+    issuerName LegalEntity (1..1) <"Specifies the issuing entity name or LEI.">
 
-    collateralAssetType AssetType (0..*) <"Represents a filter based on the asset product type.">
-    assetCountryOfOrigin ISOCountryCodeEnum (0..*) <"Represents a filter on the asset country of origin based on the ISO Standard 3166.">
-    denominatedCurrency CurrencyCodeEnum (0..*) <"Represents a filter on the underlying asset denominated currency based on ISO Standards.">
-    agencyRating AgencyRatingCriteria (0..*) <"Represents an agency rating based on default risk and creditors claim in event of default associated with specific instrument.">
-    maturityType MaturityTypeEnum (0..1) <"Specifies whether the maturity range is the remaining or original maturity.">
-    maturityRange PeriodRange (0..1) <"Represents a filter based on the underlying asset maturity.">
-    specificAssets Asset (0..*) <"Represents a filter based on specifically identified assets (eg certain securities, loans, or commodities, etc), including using source identifiers such as CUSIP or ISIN.">
-    collateralTaxonomy CollateralTaxonomy (0..*) <"Specifies the collateral taxonomy,which is composed of a taxonomy value and a taxonomy source.">
-    domesticCurrencyIssued boolean (0..1) <"Identifies that the Security must be denominated in the domestic currency of the issuer.">
-    listing ListingType (0..1) <"Specifies the exchange, index or sector specific to listing of a security.">
+type IssuerAgencyRating:
+    issuerAgencyRating AgencyRatingCriteria (1..1) <"Represents an agency rating based on default risk and creditors claim in event of default associated with asset issuer.">
 
-    condition AssetCriteriaChoice: <"If any are specified, only one of AssetType, CollateralTaxonomy or Asset should exist.">
-        optional choice collateralAssetType, collateralTaxonomy, specificAssets
+type SovereignAgencyRating:
+    sovereignAgencyRating AgencyRatingCriteria (1..1) <"Represents an agency rating based on default risk of the country of the issuer.">
 
-type ListingType: <"Specifies a filter based on an underlying corporate financial official listing defined at a stock exchange.">
+type AssetAgencyRating:
+    assetAgencyRating AgencyRatingCriteria (1..1) <"Represents an agency rating based on default risk and creditors claim in event of default associated with specific instrument.">
+
+type AssetMaturity:
+    maturityType MaturityTypeEnum (1..1) <"Specifies whether the maturity range is the remaining or original maturity.">
+    maturityRange PeriodRange (1..1) <"Represents a filter based on the underlying asset maturity.">
+
+type ListingExchange: <"Specifies a filter based on a stock exchange.">
```

Comment:
`ListingExchange` is included in the new `CollateralCriteria` choice statement. Does the cardinality of `exchange` not need to be (1..*), otherwise you can choose `ListingExchange` but not actually put one in?

Reply:
updated